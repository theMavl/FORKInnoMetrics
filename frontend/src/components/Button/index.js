import React from 'react'
import PropTypes from 'prop-types'
import styles from './style.css'

class Button extends React.Component {
  render() {
    const {name, icon, styleType, style, disabled, ...restProps} = this.props
    return (
        <button className={styles[styleType]}
                style={style}
                disabled={disabled}
                {...restProps}
        >
          <i className={`${'material-icons'}`}
          >
            {icon}
          </i>
          <span>{name}</span>
        </button>

    )
  }
}

Button.propTypes = {
  name: PropTypes.string.isRequired,
  styleType: PropTypes.oneOf(['primary', 'secondary', 'action']),
  disabled: PropTypes.bool,
  style: PropTypes.object
}

Button.defaultProps = {
  styleType: 'primary',
  disabled: false,
  style: {}
}

export default Button