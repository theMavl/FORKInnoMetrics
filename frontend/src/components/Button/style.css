.button {
    min-width: calc(var(--cell-size) * 3);
    min-height: calc(var(--cell-size) * 0.75);
    border: none;
    border-radius: 5px;
    font-size: larger;
    box-shadow: 0 0 5px 0px rgba(100, 100, 100, 0.5);

    display: flex;
    align-items: center;
    justify-content: center;

    cursor: pointer;
}

.button:disabled {
    filter: brightness(75%);
}

.button:active {
    filter: brightness(115%);
}

/*.button:focus {*/
    /*outline: none;*/
/*}*/

.button > i {
    padding-right: 5px;
    font-weight: bold;
}

.primary {
    composes: button;
    background-color: rgba(30, 235, 145, 1);
    color: rgba(255, 255, 255, 1);
}

.secondary {
    composes: button;
    background-color: rgba(255, 255, 255, 1);
    color: rgba(60, 200, 255, 1);
    border: rgba(60, 200, 255, 1) solid 1px;
}

.action {
    composes: button;
    background-color: rgba(255, 255, 255, 1);
    /*color: rgba(100, 100, 100, 1);*/
    /*border: rgba(100, 100, 100, 1) solid 1px;*/
}