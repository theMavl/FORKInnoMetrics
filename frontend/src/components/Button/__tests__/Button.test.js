import React from 'react'
import TestRenderer from 'react-test-renderer'
import Button from '../index'

describe('Test Button component', () => {

  it('Test render Button component', () => {

    const props = {
      name: 'Button',
      styleType: 'primary',
      disabled: false
    }

    const instance = TestRenderer.create(
      <Button {...props}/>
    )
    // expect(instance.toJSON().children[0]).toEqual(props.name)
    expect(instance.toJSON()).toMatchSnapshot()
  })
})