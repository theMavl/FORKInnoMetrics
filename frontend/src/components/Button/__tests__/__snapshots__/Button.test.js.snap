// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component Test render Button component 1`] = `
<button
  disabled={false}
  style={Object {}}
>
  <i
    className="material-icons"
  />
  <span>
    Button
  </span>
</button>
`;
